front/lib/actions/mcp_helper.ts

```diff
@@ -99,29 +99,35 @@ export function isRemoteMCPServerType(
 
 export function getMcpServerViewDisplayName(
   view: MCPServerViewType,
-  action?: AssistantBuilderActionConfiguration
+  action?: AssistantBuilderMCPActionConfiguration
 ) {
   return getMcpServerDisplayName(view.server, action);
 }
 
 export function getMcpServerDisplayName(
   server: MCPServerType,
-  action?: AssistantBuilderActionConfiguration
+  action?: AssistantBuilderMCPActionConfiguration
 ) {
   // Unreleased internal servers are displayed with a suffix in the UI.
   const res = getInternalMCPServerNameAndWorkspaceId(server.sId);
   let displayName = asDisplayName(server.name);
 
   if (res.isOk()) {
-    const serverConfig = INTERNAL_MCP_SERVERS[res.value.name];
-
-    if (serverConfig.isRestricted !== undefined) {
-      displayName += " (Preview)";
+    // If there is a custom name, add it to the display name (except run_dust_app, which is handled below).
```

Comment:
Do we want to show as display name or as it is? Not sure how to handle for dust_app 🤔